cpp/epidemiology/utils/eigen_util.h

```diff
@@ -242,7 +241,7 @@ class RowMajorIterator
             return &value;
         }
     };
-    using pointer = std::conditional_t<details::IsElementReference<MatrixReference>::value,
+    using pointer = std::conditional_t<details::IsElementReference<MatrixRef>::value,
```

Comment:
Not a big issue, I was just wondering: Is `MatrixRef` still needed? As far as I can tell it is only used here and in the definition of the ctor. Here I believe we can use `MatrixPtr` as well, and for the ctor just `M&`?

Reply:
MatrixRef does contain const-ness. A const iterator shouldn't require a non-const reference in the constructor and it's possible that const `operator()` returns a value where non-const `operator()` returns a reference.